Issue #203 Add Support for comparison of DateTimeOffset against a str…

drGarbinsky opened the pull request:
Issue #203 Add Support for comparison of DateTimeOffset against a string constant 

Motivation
----------
Comparing DateTimeOffset values to a constant string value in a where clause will result in an unquoted string.

Modifications
-------------
-Add an ORed expression to an if statement in N1QlExpressionTreeVisitor.VisitConstant

Results
-------
LINQ Where clause can properly quote constants when compared to DateTimeOffset properties.

MikeGoldsmith commented:
Hi @drGarbinsky 
We actually saw and discussed your work not long before you created the PR. I think this is a good improvement.
I also don't think the AppVeyor failure is down to this change as it looks like it failed to download file before building. I'll look more into that.

Thanks

drGarbinsky commented:
I'll take a crack at it today. The recommended change makes sense.

drGarbinsky commented:
Updated the PR, Let me know if I approached the change correctly. Added a DateTimeOffset property for Beer which now diverges from the sample data in terms of schema. Not sure if that's the correct approach.

brantburnett commented on Src/Couchbase.Linq.NetStandard/UnixMillisecondsDateTimeOffset.cs:
```diff
@@ -0,0 +1,107 @@
+﻿using System;
```
Instead of repeating this class for NetStandard, it should just be added as a Link to the file in the other project.  This must be done by manually editing the csproj file in a text editor.

drGarbinsky commented:
File Linked

brantburnett commented:
@MikeGoldsmith 

This looks good to me if you want to merge it.

Brant
